sdks/java/io/hcatalog/src/test/java/org/apache/beam/sdk/io/hcatalog/HCatalogIOTest.java

```diff
@@ -0,0 +1,342 @@
+  IOException, ClassNotFoundException {
+    thrown.expectCause(isA(UserCodeException.class));
+    thrown.expectMessage(containsString("org.apache.hive.hcatalog.common.HCatException"));
+    thrown.expectMessage(containsString("NoSuchObjectException"));
+    pipeline.apply(
+        Create.of(
+            toHCatRecord(0), toHCatRecord(1), toHCatRecord(2), toHCatRecord(3),
+            toHCatRecord(4), toHCatRecord(5), toHCatRecord(6), toHCatRecord(7),
+            toHCatRecord(8), toHCatRecord(9), toHCatRecord(10), toHCatRecord(11),
+            toHCatRecord(12), toHCatRecord(13), toHCatRecord(14), toHCatRecord(15),
+            toHCatRecord(16), toHCatRecord(17), toHCatRecord(18), toHCatRecord(19),
+            toHCatRecord(20), toHCatRecord(21), toHCatRecord(22), toHCatRecord(23),
+            toHCatRecord(24), toHCatRecord(25), toHCatRecord(26), toHCatRecord(27),
+            toHCatRecord(28), toHCatRecord(29), toHCatRecord(30), toHCatRecord(31),
+            toHCatRecord(32), toHCatRecord(33), toHCatRecord(34), toHCatRecord(35),
+            toHCatRecord(36), toHCatRecord(37), toHCatRecord(38), toHCatRecord(39))
+        )
+        .apply(HCatalogIO.write()
+            .withConfigProperties(getConfigProperties())
+            .withTable("myowntable"));
+    pipeline.run();
+  }
+
+  /**
+   * Test of Write without specifying a table.
+   */
+  @Test
+  public void testWriteFailureValidationTable() throws CommandNeedRetryException,
+  IOException, ClassNotFoundException {
+    thrown.expect(NullPointerException.class);
+    thrown.expectMessage(containsString("table"));
+    pipeline.apply(
+        Create.of(
+            toHCatRecord(0), toHCatRecord(1), toHCatRecord(2), toHCatRecord(3),
+            toHCatRecord(4), toHCatRecord(5), toHCatRecord(6), toHCatRecord(7),
+            toHCatRecord(8), toHCatRecord(9), toHCatRecord(10), toHCatRecord(11),
+            toHCatRecord(12), toHCatRecord(13), toHCatRecord(14), toHCatRecord(15),
+            toHCatRecord(16), toHCatRecord(17), toHCatRecord(18), toHCatRecord(19),
+            toHCatRecord(20), toHCatRecord(21), toHCatRecord(22), toHCatRecord(23),
+            toHCatRecord(24), toHCatRecord(25), toHCatRecord(26), toHCatRecord(27),
+            toHCatRecord(28), toHCatRecord(29), toHCatRecord(30), toHCatRecord(31),
+            toHCatRecord(32), toHCatRecord(33), toHCatRecord(34), toHCatRecord(35),
+            toHCatRecord(36), toHCatRecord(37), toHCatRecord(38), toHCatRecord(39))
+        )
+        .apply(HCatalogIO.write()
+            .withConfigProperties(getConfigProperties()));
+    pipeline.run();
+  }
+
+  /**
+   * Test of Write without specifying configuration properties.
+   */
+  @Test
+  public void testWriteFailureValidationConfigProp() throws CommandNeedRetryException,
+  IOException, ClassNotFoundException {
+    thrown.expect(NullPointerException.class);
+    thrown.expectMessage(containsString("configProperties"));
+    pipeline.apply(
+        Create.of(
```

Comment:
Same

Actually, for these tests that verify that the transform itself is malformed, you don't even need the `TestPipeline`. You can just do `HCatalogIO.write().withTable("myowntable").validate(null)`.